fenix/app/src/main/java/org/mozilla/fenix/compose/list/ListItem.kt

```diff
@@ -200,6 +206,54 @@ fun IconListItem(
     )
 }
 
+/**
+ * List item used to display a label with an optional description text and
+ * an [RadioButton] at the beginning.
+ *
+ * @param label The label in the list item.
+ * @param modifier [Modifier] to be applied to the layout.
+ * @param description An optional description text below the label.
+ * @param maxDescriptionLines An optional maximum number of lines for the description text to span.
+ * @param onClick Called when the user clicks on the item.
+ * @param selected [Boolean] That indicates whether the [RadioButton] is currently selected.
+ */
+@Composable
+fun RadioButtonListItem(
+    label: String,
+    modifier: Modifier = Modifier,
+    description: String? = null,
+    maxDescriptionLines: Int = 1,
+    onClick: (() -> Unit),
+    selected: Boolean = false,
+) {
+    ListItem(
+        label = label,
+        modifier = modifier
+            .clearAndSetSemantics {
+                this.selected = selected
+                role = Role.RadioButton
+                contentDescription = if (description != null) {
+                    "$label.$description"
```

Comment:
How did you come up with this content description? Should we be doing something similar with the other list items?

Reply:
https://www.figma.com/file/GT6oZLFw32LiCwKl5GOMWD/Full-Page-Translation?node-id=7505%3A18997&mode=dev

Reply:
Should there be a space between the label and the description then? (I admit I haven't tested the talkback locally)

Reply:
Because it should be a pause between title and description, otherwise, the talkback will read it as one sentence.